src/token/Rift.sol

```diff
@@ -1,58 +1,100 @@
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() initializer {}
 
     function initialize(address _owner) public initializer {
         owner = _owner;
         __ERC20_init("RIFT", "Rift Token");
+        __ERC20Burnable_init();
+        __AccessControl_init();
+        __ERC20Permit_init("Rift Token");
+        _grantRole(DEFAULT_ADMIN_ROLE, owner);
     }
 
     modifier onlyOwner() {
         require(msg.sender == owner, "ONLY_OWNER");
         _;
     }
 
-    modifier onlyMinter() {
-        require(isMinter[msg.sender], "ONLY_MINTER");
-        _;
-    }
+    // ----------- Ownership -----------
 
-    modifier onlyBurner() {
-        require(isBurner[msg.sender], "ONLY_BURNER");
-        _;
+    /// @dev Init transfer of ownership of the contract to a new account (`_pendingOwner`).
+    /// @param _pendingOwner pending owner of contract
+    /// Can only be called by the current owner.
+    function transferOwnership(address _pendingOwner) external onlyOwner {
+        pendingOwner = _pendingOwner;
+        // important to grant admin role to pendingOwner here
+        _grantRole(DEFAULT_ADMIN_ROLE, pendingOwner);
+        emit OwnershipTransferInitiated(owner, pendingOwner);
     }
 
-    function addMinter(address _newMinter) public onlyOwner {
-        require(!isMinter[_newMinter], "ALREADY_MINTER");
-        isMinter[_newMinter] = true;
+    /// @dev Accept transfer of ownership of the contract.
+    /// Can only be called by the pendingOwner.
+    function acceptOwnership() external {
+        require(msg.sender == pendingOwner, "ONLY_PENDING_OWNER");
+        address oldOwner = owner;
+        owner = pendingOwner;
+
+        // revoke the DEFAULT ADMIN ROLE from prev owner
+        _revokeRole(DEFAULT_ADMIN_ROLE, oldOwner);
+
+        emit OwnershipTransferred(oldOwner, owner);
     }
 
-    function addBurner(address _newBurner) public onlyOwner {
-        require(!isBurner[_newBurner], "ALREADY_BURNER");
-        isBurner[_newBurner] = true;
+    // ----------- Mint / Burn -----------
+    // note: users can burn their tokens
+
+    /// @dev Mint tokens.
+    /// Can only be called by MINTER_ROLE.
+    function mint(address account, uint256 amount) public onlyRole(MINTER_ROLE) {
+        _mint(account, amount);
     }
 
-    function revokeMinter(address _oldMinter) public onlyOwner {
-        require(isMinter[_oldMinter], "NOT_MINTER");
-        isMinter[_oldMinter] = false;
+    /// @dev Burn tokens from a given account.
+    /// Can only be called by BURNER_ROLE.
+    function burnFrom(address account, uint256 amount) public override onlyRole(BURNER_ROLE) {
+        super.burnFrom(account, amount);
     }
 
-    function revokeBurner(address _oldBurner) public onlyOwner {
-        require(isBurner[_oldBurner], "NOT_BURNER");
-        isBurner[_oldBurner] = false;
+    // ----------- The following functions are overrides required by Solidity. -----------
```

Comment:
ah makes sense, want to add a comment to indicate that it's to ensure that the votes contract can write its checkpoints / move voting power

Reply:
we can actually get rid of these overrides if we don't inherit from `ERC20BurnableUpgradeable` and implement it's methods ourselves.

Reply:
this seems reasonable to me. will comment on your notes below